plugins/woocommerce/includes/admin/settings/class-wc-settings-payment-gateways-react.php

```diff
@@ -96,16 +98,56 @@ private function should_render_react_section( $section ) {
 	 * Render the React section.
 	 *
 	 * @param string $section The section to render.
+	 * @param array  $payment_gateways Payment gateways.
 	 */
-	private function render_react_section( $section ) {
+	private function render_react_section( string $section ) {
 		global $hide_save_button;
 		$hide_save_button = true;
 		echo '<div id="experimental_wc_settings_payments_' . esc_attr( $section ) . '"></div>';
 
 		// Output the gateways data to the page so the React app can use it.
 		$controller = new WC_REST_Payment_Gateways_Controller();
 		$response   = $controller->get_items( new WP_REST_Request( 'GET', '/wc/v3/payment_gateways' ) );
-		echo '<script type="application/json" id="experimental_wc_settings_payments_gateways">' . wp_json_encode( $response->data ) . '</script>';
+		$payment_gateways = $this->format_payment_gateways_for_output( $response->data );
+
+		$is_woopayments_onboarded = WooCommercePayments::is_connected() && ! WooCommercePayments::is_account_partially_onboarded();
+		$is_woopayments_in_test_mode = $is_woopayments_onboarded &&
+			method_exists( WC_Payments::class, 'mode' ) &&
+			method_exists( WC_Payments::mode(), 'is_test_mode_onboarding' ) &&
+			WC_Payments::mode()->is_test_mode_onboarding();
+
+		// TODO: we should think about a better way to pass this data to the frontend.
+		echo '<script type="application/json" id="experimental_wc_settings_payments_woopayments">' . wp_json_encode(
+			array(
+				'isSupported'  => WooCommercePayments::is_supported(),
+				'isOnboarded'  => $is_woopayments_onboarded,
+				'isInTestMode' => $is_woopayments_in_test_mode,
+			)
+		) . '</script>';
+		echo '<script type="application/json" id="experimental_wc_settings_payments_gateways">' . wp_json_encode( $payment_gateways ) . '</script>';
+	}
+
+	/**
+	 * Handle some additional formatting and processing that is necessary to display gateways on the React settings page.
+	 *
+	 * @param array $payment_gateways The payment gateways.
+	 *
+	 * @return array
+	 */
+	private function format_payment_gateways_for_output( array $payment_gateways ): array {
+		$offline_methods = [ 'bacs', 'cheque', 'cod' ];
+		$display_payment_gateways = [];
+
+		// Remove offline methods from the list of gateways (these are handled differently).
```

Comment:
hmm, maybe we should handle it differently and maybe not - we still need this data passed to the client
for now offline methods are hardcoded, but we could sort it out in a different issue!

Reply:
Agreed, let's leave that for a different issue!